game-core/src/main/java/games/strategy/triplea/attachments/CanalAttachment.java

```diff
@@ -154,6 +160,22 @@ public void validate(final GameData data) throws GameParseException {
     if (m_landTerritories == null || m_landTerritories.size() == 0) {
       throw new GameParseException("Canal named " + m_canalName + " must have landTerritories set!" + thisErrorMsg());
     }
+    final Set<Territory> territories = new HashSet<>();
+    for (final Territory t : data.getMap()) {
+      final Set<CanalAttachment> canalAttachments = get(t);
+      if (canalAttachments.isEmpty()) {
+        continue;
+      }
```

Comment:
It's late, so I may not be reading this properly, but is this `if/continue` necessary?  If `canalAttachments` is empty, won't the for-each loop below simply do nothing?  That is, is the following equivalent?

```java
    final Set<Territory> territories = new HashSet<>();
    for (final Territory t : data.getMap()) {
      for (final CanalAttachment canalAttachment : get(t)) {
        if (canalAttachment.getCanalName().equals(m_canalName)) {
          territories.add(t);
        }
      }
    }
```

**EDIT:** I see now that this is legacy code that was just copy/pasted, so if you don't feel comfortable changing it, don't worry about it.

Reply:
Yeah, didn't touch this code but I'm fine making that small improvement. Done.